src/libraries/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.DivRem.cs

```diff
@@ -72,64 +67,55 @@ public static uint Remainder(uint[] left, uint right)
             return (uint)carry;
         }
 
-        public static unsafe uint[] Divide(uint[] left, uint[] right,
-                                           out uint[] remainder)
+        public static uint[] Divide(ReadOnlySpan<uint> left, ReadOnlySpan<uint> right, out uint[] remainder)
         {
-            Debug.Assert(left != null);
-            Debug.Assert(right != null);
             Debug.Assert(left.Length >= 1);
             Debug.Assert(right.Length >= 1);
             Debug.Assert(left.Length >= right.Length);
 
-            // Switching to unsafe pointers helps sparing
-            // some nasty index calculations...
-
             // NOTE: left will get overwritten, we need a local copy
 
-            uint[] localLeft = CreateCopy(left);
+            uint[] localLeft = left.ToArray();
             uint[] bits = new uint[left.Length - right.Length + 1];
 
-            fixed (uint* l = &localLeft[0], r = &right[0], b = &bits[0])
-            {
-                Divide(l, localLeft.Length,
-                       r, right.Length,
-                       b, bits.Length);
-            }
+            Divide(localLeft, right, bits);
 
             remainder = localLeft;
 
             return bits;
         }
 
-        public static unsafe uint[] Divide(uint[] left, uint[] right)
+        public static uint[] Divide(ReadOnlySpan<uint> left, ReadOnlySpan<uint> right)
         {
-            Debug.Assert(left != null);
-            Debug.Assert(right != null);
             Debug.Assert(left.Length >= 1);
             Debug.Assert(right.Length >= 1);
             Debug.Assert(left.Length >= right.Length);
 
             // Same as above, but only returning the quotient.
 
-            // NOTE: left will get overwritten, we need a local copy
-
-            uint[] localLeft = CreateCopy(left);
             uint[] bits = new uint[left.Length - right.Length + 1];
 
-            fixed (uint* l = &localLeft[0], r = &right[0], b = &bits[0])
+            // NOTE: left will get overwritten, we need a local copy
+            // However, mutated left is not used afterwards, so use array pooling or stack alloc
+            if (left.Length < AllocationThreshold)
             {
-                Divide(l, localLeft.Length,
-                       r, right.Length,
-                       b, bits.Length);
+                Span<uint> leftCopy = stackalloc uint[left.Length];
+                left.CopyTo(leftCopy);
+                Divide(leftCopy, right, bits);
+            }
+            else
+            {
+                uint[] leftCopy = ArrayPool<uint>.Shared.Rent(left.Length);
+                left.CopyTo(leftCopy);
+                Divide(new Span<uint>(leftCopy, 0, left.Length), right, bits);
```

Comment:
```suggestion
                Divide(leftCopy.AsSapn(0, left.Length), right, bits);
```
Or "merge" it with the stack-alloc branch.